Sources/protoc-gen-swift/MessageFieldGenerator.swift

```diff
@@ -198,7 +198,7 @@ class MessageFieldGenerator: FieldGeneratorBase, FieldGenerator {
 
         let conditional: String
         if isRepeated {  // Also covers maps
-            conditional = "!\(varName).isEmpty"
+            conditional = "!\(varName).isEmpty || visitor.shouldIncludeDefault()"
         } else if hasFieldPresence {
             conditional = "let v = \(storedProperty)"
```

Comment:
I am not so sure whether we need to handle this. Looks like in proto3, only submessage uses it. Most of fields in proto2 using it. If we also want to get nil in JsonFormat, we can write another method call visitNull(fieldNumber: Int) throws

Reply:
I haven't looked at the other languages, do they really generate an empty list in json, that seems odd?

Reply:
I verified on python and go, if it is empty list, they output nothing.